pkg/lang/javascript/plugin_persist.go

```diff
@@ -168,12 +168,11 @@ func (p *persister) handleFile(f *core.SourceFile, unit *core.ExecutionUnit) ([]
 			continue
 		}
 
-		if annot.Capability.Directives.Object(core.EnvironmentVariableDirective) != nil {
-			continue
-		}
-
 		keyType, pResult := p.determinePersistType(f, annot)
 		if pResult == nil {
+			if annot.Capability.Directives.Object(core.EnvironmentVariableDirective) != nil {
```

Comment:
```suggestion
			if len(annot.Capability.Directives.Object(core.EnvironmentVariableDirective)) == 0 {
```
This will work for both nil and empty.

Reply:
it doesnt let you do len on that object